web/pgadmin/browser/static/js/browser.js

```diff
@@ -1650,6 +1651,10 @@ define('pgadmin.browser', [
 
     },
 
+    disableTreeSelection: function(_disable) {
+      pgBrowser.disable_tree_select = _disable;
```

Comment:
I think disable selection value should be internal to tree object which can internally check when calling .select. That way we will not need to add check at all the places.
Other than that, I don't think this is a good solution. You're disable select on search objects - kind of a hack and not solution.